mail-client/mutt-wizard: added mutt-wizard with dated ebuild

epsilon-0 opened the pull request:
Automatic configuration creator and account manager for neomutt.

Really helpful for people who want to learn mutt but are daunted by setting it up.

Signed-off-by: Aisha Tammy <gentoo@aisha.cc>

AndrewAmmerlaan commented:
Hi there,

Sometimes 'new package' PRs get snowed under by all the other ones (new packages are low priority compared to fixes to already existing packages). Therefore it might take quite some time for this to get any attention. If you'd like, you could also make this package available to Gentoo end-users through the [GURU overlay](https://wiki.gentoo.org/wiki/Project:GURU). This is an ebuild repository maintained collaboratively by Gentoo users. To gain access to the repository, simply create an "Access request" on bugs.gentoo.org.

juippis commented on mail-client/mutt-wizard/mutt-wizard-20200209.ebuild:
```diff
@@ -0,0 +1,48 @@
+
+LICENSE="GPL-3"
+SLOT="0"
+IUSE="+notmuch links +w3m lynx +libnotify +abook +gpg +cron +imagemagick"
```
This looks like a series of shell scripts with no configurability during build-time, hence using USE flags here is banned.
https://projects.gentoo.org/qa/policy-guide/dependencies.html#pg0001

epsilon-0 replied:
I have removed the unneeded flags

juippis commented on mail-client/mutt-wizard/mutt-wizard-20200209.ebuild:
```diff
@@ -0,0 +1,48 @@
+DEPEND="mail-client/neomutt:=[notmuch=]
+	net-mail/isync
+	mail-mta/msmtp
+	app-admin/pass
```
With nothing to compile, this list looks wrong. Maybe they belong to RDEPEND?

juippis commented on mail-client/mutt-wizard/mutt-wizard-20200209.ebuild:
```diff
@@ -0,0 +1,48 @@
+	w3m? ( www-client/w3m )
+	libnotify? ( x11-libs/libnotify )
+	abook? ( app-misc/abook )
+	gpg? ( app-crypt/gnupg )
```
As explained, these have no function in the ebuild, and shouldn't be used. Use `optfeature` from *eutils.eclass* instead.

epsilon-0 replied:
I have removed them.

juippis commented on mail-client/mutt-wizard/mutt-wizard-20200209.ebuild:
```diff
@@ -0,0 +1,48 @@
+"
+
+BDEPEND="net-mail/isync
+	app-admin/pass
```
With no unpack or src_compile defined, this is most likely also wrong.

epsilon-0 replied:
Indeed, I have moved them to RDEPEND.

epsilon-0 commented:
@juippis thanks a lot for your comments, I really appreciate them.
I'll work on fixing them asap and get back to you.

Much regards.

epsilon-0 commented:
@juippis I've made all the changes that you have recommended.
Thanks a lot again. I'm learning a lot and appreciate your patience and help :)

Much regards.

thesamesam commented on mail-client/mutt-wizard/mutt-wizard-20200313.ebuild:
```diff
@@ -0,0 +1,30 @@
+RDEPEND="
+  mail-client/neomutt:=[notmuch=]
+  net-mail/isync
+	mail-mta/msmtp
```
Indentation

epsilon-0 replied:
Fixed.

thesamesam commented on mail-client/mutt-wizard/mutt-wizard-20200313.ebuild:
```diff
@@ -0,0 +1,30 @@
+  mail-client/neomutt:=[notmuch=]
+  net-mail/isync
+	mail-mta/msmtp
+	app-admin/pass
```
Indentation

epsilon-0 replied:
I've checked the install script and it is indeed only needed during runtime.

The README needs to be changed.

thesamesam commented on mail-client/mutt-wizard/mutt-wizard-20200313.ebuild:
```diff
@@ -0,0 +1,30 @@
+
+RDEPEND="
+  mail-client/neomutt:=[notmuch=]
+  net-mail/isync
```
Is this just a runtime dependency? The README says it's needed for install.

epsilon-0 replied:
I've checked the install script and it is indeed only needed during runtime.

The README needs to be changed.

And indeed, I will add `ssl` to isync.

I really appreciate your help

juippis commented on mail-client/mutt-wizard/mutt-wizard-20200313.ebuild:
```diff
@@ -0,0 +1,30 @@
+
+LICENSE="GPL-3"
+SLOT="0"
+IUSE="+notmuch +lynx"
```
These USE flags still have no function in this ebuild.

epsilon-0 replied:
done

juippis commented on mail-client/mutt-wizard/mutt-wizard-20200313.ebuild:
```diff
@@ -0,0 +1,30 @@
+IUSE="+notmuch +lynx"
+
+RDEPEND="
+	mail-client/neomutt:=[notmuch=]
```
`[notmuch=]` looks weird. Take off the `=`.

epsilon-0 replied:
done

juippis commented on mail-client/mutt-wizard/mutt-wizard-20200313.ebuild:
```diff
@@ -0,0 +1,30 @@
+	net-mail/isync:=[ssl]
+	mail-mta/msmtp
+	app-admin/pass
+	lynx? ( www-client/lynx )
```
Banned, add as `optfeature` from *eutils.eclass* instead.

epsilon-0 replied:
added this, plus others

thesamesam commented on mail-client/mutt-wizard/mutt-wizard-20200430.ebuild:
```diff
@@ -0,0 +1,40 @@
+
+LICENSE="GPL-3"
+SLOT="0"
+IUSE=""
```
Drop

epsilon-0 replied:
done

thesamesam commented on mail-client/mutt-wizard/mutt-wizard-20200430.ebuild:
```diff
@@ -0,0 +1,40 @@
+
+# needed because there is no 'all' target defined in MAKEFILE
+src_compile() {
+	return 0;
```
I think people normally do :; or something like that, but I guess this works.

epsilon-0 replied:
i think this is a tiny bit more verbose, which is never a problem.

thesamesam commented on mail-client/mutt-wizard/mutt-wizard-20200430.ebuild:
```diff
@@ -0,0 +1,40 @@
+	return 0;
+}
+
+pkg_postinst() {
```
Maybe only show this on new installs (check ${REPLACING_VERSIONS})?

epsilon-0 replied:
might be good to keep it for everyone, as it changes a fair over time, by adding a lot of features.

juippis commented on mail-client/mutt-wizard/mutt-wizard-20200430.ebuild:
```diff
@@ -0,0 +1,39 @@
+SLOT="0"
+
+RDEPEND="
+	mail-client/neomutt:=[notmuch]
+	net-mail/isync:=[ssl]
```
Since this does no compiling, no libraries can be linked. (Take out `:=`)

epsilon-0 replied:
Done.

juippis commented on mail-client/mutt-wizard/mutt-wizard-20200430.ebuild:
```diff
@@ -0,0 +1,39 @@
+SLOT="0"
+
+RDEPEND="
+	mail-client/neomutt:=[notmuch]
+	net-mail/isync:=[ssl]
+	mail-mta/msmtp
+	app-admin/pass
```
Sort this list alphabetically.

epsilon-0 replied:
Done.

epsilon-0 commented:
> You'll need to describe the package in your git commit **message** box. Just a sentence or two is enough. Sorry I failed to mention this the first time.

I've added a long description in the metadata and also added that in the commit message.
Hopefully that is fine.

> 
> Now to the biggest problem:
> 
> ```
>  * The ebuild is installing to one or more unexpected paths:
>  * 
>  *   /usr/local
>  * 
>  * Please fix the ebuild to use correct FHS/Gentoo policy paths.
> ```

Tots forgot about this. I had it fixed in my overlay but forgot to add it here :stuck_out_tongue:  
Have now fixed it here.

Thanks a lot for all your pointers.

juippis commented on mail-client/mutt-wizard/Manifest:
```diff
@@ -0,0 +1 @@
+DIST mutt-wizard-20200430.tar.gz 32140 BLAKE2B 37656f1a25eecf5caa982dc898e0c3fc45b38b1c3d2db8673ecdf1a3694d3c3a856239830bdc86182b97e84b4c18d7ed42720be4ec60125cc5060817df6ff262 SHA512 aa653421a3492c19a88e0ed421f554d7e0af26d58c7420887b0be2b9db6f481cd439da359af8207fd11fb7180b289535038ad0ce94700c8123c84445aa83ed90
```
Now I don't know if no one spotted this already or if you changed it, but we name packages like these `mutt-wizard-0_pre20200430.ebuild`. Then when whatever release is made, it will always be an update. Now 20200430 -> 1.0 would be a downgrade.

Rest looks good!

epsilon-0 replied:
lmao, wow, I didn't even know that.  
I didn't change it, will fix it :smile_cat:

epsilon-0 replied:
done

titanofold commented:
So, is it mutt or neomutt that this tool is for? The DEPEND makes it seem like it's neomutt and not mutt. Also, the details I read about it indicate neomutt and not mutt.

I know you copy/pasted the description, which is fine. However, in this case it seems disingenuous.

epsilon-0 commented:
> So, is it mutt or neomutt that this tool is for? The DEPEND makes it seem like it's neomutt and not mutt. Also, the details I read about it indicate neomutt and not mutt.
> 
> I know you copy/pasted the description, which is fine. However, in this case it seems disingenuous.

A valid point and I couldn't come up with a good solution :confused:  
This is an unfortunate situation but there isn't too much to do on this side.

Any other option is welcome.

epsilon-0 commented:
> So, is it mutt or neomutt that this tool is for? The DEPEND makes it seem like it's neomutt and not mutt. Also, the details I read about it indicate neomutt and not mutt.
> 
> I know you copy/pasted the description, which is fine. However, in this case it seems disingenuous.

To clarify, this tool is for neomutt.

titanofold commented:
Then I think the description should be edited to say as much.

epsilon-0 commented:
> Then I think the description should be edited to say as much.

done

Zlogene commented on mail-client/mutt-wizard/mutt-wizard-0_pre20200430.ebuild:
```diff
@@ -0,0 +1,44 @@
+SLOT="0"
+
+RDEPEND="
+	app-admin/pass
```
this must be at least altered to also provide gopass.

epsilon-0 replied:
Do you mean that there should be a choice?

Zlogene replied:
yes, as pass is kinda obsolet thingy and gopass has backward compatibility in anyway.

epsilon-0 replied:
Ah, looked it up, seems like gopass does not add a `pass` command so it is unlikely to work as of now.  
Unless upstream adds an option to check for gopass it can't be added.  
Will create an issue with upstream.

epsilon-0 replied:
upstream issue has been created: https://github.com/LukeSmithxyz/mutt-wizard/issues/433

titanofold commented on mail-client/mutt-wizard/mutt-wizard-0_pre20200430.ebuild:
```diff
@@ -0,0 +1,44 @@
+
+inherit eutils
+
+DESCRIPTION="A system for automatically configuring mutt and isync"
```
This should say neomutt, too

epsilon-0 replied:
ooof, I keep forgetting this.  
thanks for noticing :smile_cat:

epsilon-0 commented:
Am also going to add a live ebuild. Given that the project is not making releases and the build system is staying pretty much the same.

juippis commented:
I'm kind of tempted to wait for an upstream release, looking at previously filed issues, there seems to be a pressure to make one.

And maybe wait for upstreams response to gopass. Other than those, this looks ready to me.

epsilon-0 commented:
> I'm kind of tempted to wait for an upstream release, looking at previously filed issues, there seems to be a pressure to make one.
> 
> And maybe wait for upstreams response to gopass. Other than those, this looks ready to me.

sounds good, thanks.

epsilon-0 commented:
> I'm kind of tempted to wait for an upstream release, looking at previously filed issues, there seems to be a pressure to make one.
> 
> And maybe wait for upstreams response to gopass. Other than those, this looks ready to me.

@juippis I don't think upstream is responding. I think we can merge it.

juippis commented on mail-client/mutt-wizard/Manifest:
```diff
@@ -0,0 +1 @@
+DIST mutt-wizard-0_pre20200506.tar.gz 32139 BLAKE2B 81fb86bce808db3e4f65788d2d3a59b9a7d18f210dc29e525773f596594bc2cba42424b73a50a6a92e5271c160f4d332943b8fa9526f3d44df5da8f96ffd1dc7 SHA512 26ff620b570f1fb6a9fed31d56233a2a67055c6a3352dd547fe7c258cdf2476e47fd87478d678fce953b921d0b6ce9ab3bbec563cbc616c83db1bc5a8047bd26
```
Would you like to update this to latest available commit? I see a lot has happened in a month.

epsilon-0 replied:
Ah, yes thanks for that.

I've done the same.
